lib/inc/internal/util/agent.hpp

```diff
@@ -16,9 +16,20 @@ namespace facter { namespace util { namespace agent {
      * @param exe the name of the executable
      * @return the path to the executable
      */
-    inline std::string which(const std::string& exe) {
+inline std::string which(const std::string& exe) {
 #ifdef FACTER_PATH
-        std::string fixed = leatherman::execution::which(exe, {FACTER_PATH});
+    std::string fixed = leatherman::execution::which(exe, {FACTER_PATH}, true);
+    if (!fixed.empty()) {
+        return fixed;
+    }
+    LOG_WARNING("{1} not found at configured location {2}, using PATH instead", exe, FACTER_PATH);
```

Comment:
I think this check and logging is redundant to the one in which with 3 parameters.

Reply:
See response above